ci/scripts/pre-commit.sh

```diff
@@ -5,8 +5,14 @@ rustup default stable
 export PATH="$HOME/.local/bin:$PATH"
 PIP_BREAK_SYSTEM_PACKAGES=1 pip3 install pre-commit
 
+if [ "${CI_OVERRIDE_BUF_BREAKING:-false}" = "true" ]; then
+    PRE_COMMIT_CMD="pre-commit run"
+else
+    PRE_COMMIT_CMD="pre-commit run -a --hook-stage=manual"
+fi
+
 # Make sure CI can pull from the private repo.
-if ! SKIP=bazel_rust_format_check,bazel_smoke pre-commit run -a --hook-stage=manual; then
+if ! SKIP=bazel_rust_format_check,bazel_smoke $PRE_COMMIT_CMD; then
```

Comment:
You could also consider:

```
PRE_COMMIT_ARGS=()
if ! [ "${CI_OVERRIDE_BUF_BREAKING:-false}" = "true" ]; then
    PRE_COMMIT_ARGS+=(-a --hook-stage=manual)
fi
if ! SKIP=bazel_rust_format_check,bazel_smoke pre-commit run ${PRE_COMMIT_ARGS[@]}; then
```

Reply:
Ah okay - yeah it wasn't working quite as expected, so I switched it to just explicitly skip `buf-breaking` instead.

Reply:
Ok but then I would still consider using an array like:
```
SKIP_CHECKS=(bazel_rust_format_check,bazel_smoke)
if [ "${CI_OVERRIDE_BUF_BREAKING:-false}" = "true" ]; then
    SKIP_CHECKS+=(buf-breaking)
fi

# Make sure CI can pull from the private repo.
if ! SKIP=${SKIP_CHECKS[@]} pre-commit run -a --hook-stage=manual; then
```